src/SQL/Parser.php

```diff
@@ -44,6 +48,8 @@ final class Parser
     private const SPECIAL              = '[' . self::SPECIAL_CHARS . ']';
     private const OTHER                = '[^' . self::SPECIAL_CHARS . ']+';
 
+    private const REPLACE_PATTERNS = ['/\b([Aa][Rr][Rr][Aa][Yy])\s*\[/' => '\1['];
```

Comment:
Note that this parser is a port of the one from PDO (see the link in the class phpDoc). I'm curious if/how PDO handles this case. It would be nice if we kept the implementation close to the original.